File: Data Structures & Algorithms/binary-search/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let lo = 0
        let hi = nums.length
        console.log(hi)

        do {
            let mid = Math.floor((lo + hi) / 2)
            let guess = nums[mid]

            if (guess === target) {
                return mid
            } else if (guess > target) {
                console.log(hi)
                hi = mid - 1
                console.log(hi)
            } else {
                lo = mid + 1
            }
        } while (hi >= lo)

        return -1
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        indicies = {}

        for i, n in enumerate(nums):
            indicies[n] = i
        
        for i, n in enumerate(nums):
            diff = target - n
            if diff in indicies and indicies[diff] != i:
                return [i, indicies[diff]]
        return []